### PR TITLE
Feature bnb-native

### PR DESCRIPTION
**Pull request by BitHighlander**
(WIP)

# Add BNB Asset and BNB network support

 It enables signing of BNB tx's in native. 

Notes:

(This PR ONLY covers a transfer TX)

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: examples/sandbox/package.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
     "debug": "^4.1.1",
     "jquery": "^3.4.1",
     "json": "^9.0.6",
-    "parcel-bundler": "^1.12.4"
+    "parcel-bundler": "^1.12.4",
+    "secp256k1": "^4.0.2"
```
Comment (kaladinlight) [inline examples/sandbox/package.json:27]: Sandbox fixed? I thought we tried that, but good to see it worked. Did you ever verify that the package is a dependency in the binanceSignTx package?

Comment (BitHighlander) [reply examples/sandbox/package.json:27]: not fixed :(

Comment (BitHighlander) [reply examples/sandbox/package.json:27]: it was atom that this actually broke on. and deff the package its complaining of is already in cosmos-tx-builder ```{
  "name": "cosmos-tx-builder",
  "version": "6.3.3",
  "scripts": {
    "test": "jest"
  },
  "dependencies": {
    "bech32": "^1.1.4",
    "bip32": "^2.0.5",
    "bip39": "^3.0.2",
    "bitcoinjs-lib": "^5.1.10",
    "coininfo": "^5.1.0",
    "crypto-js": "^3.3.0",
    "dotenv": "^8.2.0",
    "secp256k1": "^3.8.0"
  }
}

Comment (BitHighlander) [reply examples/sandbox/package.json]: ill go ahead and remove as this didnt help anything

File: packages/hdwallet-native/src/binance.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      return false;
+    }
+
+    binanceGetAccountPaths(msg: core.BinanceGetAccountPaths): Array<core.BinanceAccountPath> {
+      return [
+        {
+          addressNList: [0x80000000 + 44, 0x80000000 + 117, 0x80000000 + msg.accountIdx, 0, 0],
+          // hardenedPath: [0x80000000 + 44, 0x80000000 + core.slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx],
+          // relPath: [0, 0],
+          // description: "Native",
```
Comment (kaladinlight) [inline packages/hdwallet-native/src/binance.ts:32]:
```suggestion
```

File: packages/hdwallet-native/src/native.ts [hunk 3]
```diff
@@ -138,6 +143,7 @@ export class NativeHDWallet
     await super.btcInitializeWallet(seed);
     super.ethInitializeWallet("0x" + seed.toString("hex"));
     super.cosmosInitializeWallet(this.#mnemonic);
+    super.binanceInitializeWallet(this.#mnemonic);
```
Comment (kaladinlight) [inline packages/hdwallet-native/src/native.ts:146]:
```suggestion
    super.binanceInitializeWallet(seed);
```

File: packages/hdwallet-native/src/binance.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      const message = SHA256(CryptoJS.enc.Hex.parse(publicKey));
+      const hash = RIPEMD160(message as any).toString();
+      const address = Buffer.from(hash, `hex`);
+      const cosmosAddress = this.bech32ify(address, `bnb`);
+      return cosmosAddress;
+    }
+
+    async binanceGetAddress(msg: core.BinanceGetAddress): Promise<string> {
+      const seed = await mnemonicToSeed(this.#seed);
+
```
Comment (kaladinlight) [inline packages/hdwallet-native/src/binance.ts:65]: removed in favor of using seed in `initialize`
```suggestion
```

**Pull request merged**